tools/dockerfile/distribtest/ruby_ubuntu2404_x64/Dockerfile

```
@@ -12,19 +12,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM ubuntu:20.04
+FROM ubuntu:24.04
 
 # Install Git and basic packages.
 RUN apt-get update -y && apt-get install -y \
-  curl \
-  gnupg2 \
+  curl gnupg2 \
   gcc && apt-get clean
 
+#==================
+# Ruby dependencies
+
 # Install rvm
 RUN gpg --keyserver hkp://keyserver.ubuntu.com --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
 RUN \curl -sSL https://get.rvm.io | bash -s stable
 
 # Install Ruby 3.1
+RUN /bin/bash -l -c "rvm pkg install openssl"
 RUN /bin/bash -l -c "rvm install ruby-3.1.3"
```

```suggestion
RUN /bin/bash -l -c "rvm install ruby-3.1.3"
```

I don't think this is needed now, since the Ruby install here has been bumped 3.0 ->  3.1 as of #39607 and Ruby 3.1 is linked to OpenSSL 3 by default - so should be fine against Ubuntu 24.04's default openssl without any special openssl install.

---

updated